src/executabletypes.cpp

```diff
@@ -47,4 +48,70 @@ const char *to_string(ExeSectionType type)
     static_assert(size_t(ExeSectionType::Unknown) == 2, "Enum was changed. Update switch case.");
 }
 
+void to_json(nlohmann::json &js, const ExeSymbol &d)
+{
+    js = nlohmann::json{{"name", d.name}, {"address", d.address}, {"size", d.size}};
+}
+
+void from_json(const nlohmann::json &js, ExeSymbol &d)
+{
+    js.at("name").get_to(d.name);
+    js.at("address").get_to(d.address);
+    js.at("size").get_to(d.size);
+}
+
+void to_json(nlohmann::json &js, const ExeObjectSection &d)
+{
+    js = nlohmann::json{{"name", d.name}, {"offset", d.offset}, {"size", d.size}};
+}
+
+void from_json(const nlohmann::json &js, ExeObjectSection &d)
+{
+    js.at("name").get_to(d.name);
+    js.at("offset").get_to(d.offset);
+    js.at("size").get_to(d.size);
+}
+
+void to_json(nlohmann::json &js, const ExeObject &d)
+{
+    js = nlohmann::json{{"name", d.name}, {"sections", d.sections}};
+}
+
+void from_json(const nlohmann::json &js, ExeObject &d)
+{
+    js.at("name").get_to(d.name);
+    js.at("sections").get_to(d.sections);
+}
+
+void to_json(nlohmann::json &js, const ExeImageData &d)
+{
+    js = nlohmann::json{
+        {"codealign", d.codeAlignment},
+        {"dataalign", d.dataAlignment},
+        {"codepadding", d.codePad},
+        {"datapadding", d.dataPad}};
+}
+
+void from_json(const nlohmann::json &js, ExeImageData &d)
+{
+    js.at("codealign").get_to(d.codeAlignment);
+    js.at("dataalign").get_to(d.dataAlignment);
+    js.at("codepadding").get_to(d.codePad);
+    js.at("datapadding").get_to(d.dataPad);
```

Comment:
After this change is merged, you can probably make a follow up change for both pdbreader and executable json to verify and improve error handling when a json field does not exist. I think it crashes with `at("str")` if that str field is not present in the json.